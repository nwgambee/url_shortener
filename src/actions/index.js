export const setUrls = (url) => ({
  type: 'SET_URLS',
  url
});
