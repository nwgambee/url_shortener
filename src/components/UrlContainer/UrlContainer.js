import React, { Component } from 'react';
import './UrlContainer.css';
import { connect } from 'react-redux';
import { setUrls } from '../../actions/index'
import { getUrls, deleteUrl } from '../../apiCalls'


export class UrlContainer extends Component {
  constructor() {
    super();
 }

 componentDidMount() {
  getUrls()
    .then(data => data.urls.forEach(url => {
      this.props.setUrls(url)
    }))
    .catch(err => console.error('Error fetching:', err));
}
 
 render() {
  let urlEls = []
  if (this.props.urls) {
   urlEls = this.props.urls.map(url => {
    return (
      <div className="url">
        <h3>{url.title}</h3>
        <a href={url.short_url} target="blank">{url.short_url}</a>
        <p>{url.long_url}</p>
        <button >Delete Url</button>
      </div>
    )
  });
 }
    return ( <section>
      { urlEls.length ? urlEls : <p>No urls yet! Find some to shorten!</p> }
    </section>
    )
  }
}


export const mapStateToProps = ({ urls }) => ({
  urls
});

export const mapDispatchToProps = dispatch => ({
  setUrls: (url) => dispatch(setUrls(url))
})

export default connect(mapStateToProps, mapDispatchToProps)(UrlContainer)
