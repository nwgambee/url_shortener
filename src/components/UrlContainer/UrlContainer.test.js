import { UrlContainer, mapStateToProps, mapDispatchToProps } from '../UrlContainer/UrlContainer'
import { shallow } from 'enzyme';
import React from 'react';
import { setUrls } from '../../actions/index'


describe('UrlContainer', () => {
  let wrapper;
  describe('Method Tests', () => {
    wrapper = shallow(<UrlContainer />)
    expect(wrapper).toMatchSnapshot();
  })

  describe('MapStateToProps tests', () => {
    it('should return an object with a urls property', () => {
      wrapper = shallow(<UrlContainer />)
      const mockState = {
        urls: [{longUrl: 'some url'}],
        testProp: 'this is a fake property'
      }
      const expected = {
        urls: [{longUrl: 'some url'}]
      }
      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected)

    })
  })

  describe('MapDispatchToProps tests', () => {
    it('should call dispatch with the setUrls action when setUrls is called', () => {
      const mockUrl = {longUrl: 'long url', shortUrl: 'shorter url'}
      const mockDispatch = jest.fn();
      const actionToDispatch = setUrls(mockUrl);
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.setUrls(mockUrl)

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })
  })
})