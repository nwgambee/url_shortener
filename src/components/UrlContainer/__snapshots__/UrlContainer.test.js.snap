// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UrlContainer Method Tests should match the snapshot 1`] = `
<section>
  <p>
    No urls yet! Find some to shorten!
  </p>
</section>
`;
